File: examples/PHP/hwclient.php
<?php
/*
 *  Hello World client
 *  Connects REQ socket to tcp://localhost:5555
 *  Sends "Hello" to server, expects "World" back
 * @author Ian Barber <ian(dot)barber(at)gmail(dot)com>
 */

$context = new ZMQContext();

//  Socket to talk to server
echo "Connecting to hello world server...\n";
$requester = new ZMQSocket($context, ZMQ::SOCKET_REQ);
$requester->connect("tcp://localhost:5555");

for($request_nbr = 0; $request_nbr != 10; $request_nbr++) {
	printf ("Sending request %d...\n", $request_nbr);
	$requester->send("Hello");
	
	$reply = $requester->recv();
	printf ("Received reply %d: [%s]\n", $request_nbr, $reply);
}

File: examples/Perl/hwclient.pl
#!/usr/bin/perl
=pod

Hello World client

Connects REQ socket to tcp://localhost:5555

Sends "Hello" to server, expects "World" back

Author: Daisuke Maki (lestrrat)
Original version Author: Alexander D'Archangel (darksuji) <darksuji(at)gmail(dot)com>

=cut

use strict;
use warnings;
use 5.10.0;

use ZMQ::LibZMQ3;
use ZMQ::Constants qw(ZMQ_REQ);

my $MAX_MSGLEN = 255;

my $context = zmq_init();

# Socket to talk to server
say 'Connecting to hello world server...';
my $requester = zmq_socket($context, ZMQ_REQ);
zmq_connect($requester, 'tcp://localhost:5555');

for my $request_nbr (0..9) {
    say "Sending request $request_nbr...";
    zmq_send($requester, 'Hello');
    my $message;
    my $size = zmq_recv($requester, $message, $MAX_MSGLEN);
    my $reply = substr($message, 0, $size);
    say "Received reply $request_nbr: [". $reply .']';
}
